src/boot/main.c

```diff
@@ -333,6 +333,43 @@ void check_cache_emulation() {
     __osRestoreInt(saved);
 }
 
+/**
+ * Increment the first and last values of the stack.
+ * If they're different, that means an error has occured, so trigger a crash.
+*/
+#ifdef DEBUG
+void check_stack_validity(void) {
+    gIdleThreadStack[0]++;
+    gIdleThreadStack[THREAD1_STACK - 1]++;
```

Comment:
Are _both_ the first and last bytes of the stack guaranteed to go unused when not completely full? Judging by the declaration in the thread creation, I assume the stack reads right to left, so could that be why maybe you thought it safe to change the first byte of it? Just speculation, but seems sus to me.

Reply:
yes

Reply:
Weird, why would the stack work that way as to just waste memory here?

Reply:
because I assume it starts from the middle and goes outward in both directions

Reply:
But

`create_thread(&gMainThread, THREAD_3_MAIN, thread3_main, NULL, gThread3Stack + THREAD3_STACK, 100);`

How does it know where the center of the stack is when it's only given one address?